### PR TITLE
Split ValueTypeTests for lw5 and enable basic value type tests

### PR DESCRIPTION
**Pull request by theresa-m**
Please go to the third commit to see which tests have been refactored.

Related to https://github.com/eclipse-openj9/openj9/issues/18157

### CODE REVIEW

File: test/functional/Valhalla/src_lw5/org/openj9/test/lworld/ValueTypeTests.java [hunk 1]
```diff
@@ -0,0 +1,3516 @@
...
+ * 7)  export SPEC=linux_x86-64_cmprssptrs
+ * 8)  export BUILD_LIST=functional/Valhalla
+ * 9)  export AUTO_DETECT=false
+ * 10) export JDK_IMPL=openj9
+ * 11) make -f run_configure.mk && make compile && make _sanity
+ */
+
+@Test(groups = { "level.sanity" })
+public class ValueTypeTests {
+	// static Lookup lookup = MethodHandles.lookup();
```
Comment (hangshao0) [inline test/functional/Valhalla/src_lw5/org/openj9/test/lworld/ValueTypeTests.java:45]: I see a lot of lines are commented out here. Usually we don't merge commented code. We keep the original copy of `ValueTypeTests.java` under `/src_qtypes` folder. For now, can these commented lines be removed from this file ? The deleted tests in this fie can be later replaced with newer version (with no q type dependency) gradually in the upcoming PRs.

File: runtime/vm/resolvefield.cpp [hunk 2]
```diff
@@ -1105,7 +1105,7 @@ fieldOffsetsFindNext(J9ROMFieldOffsetWalkState *state, J9ROMFieldShape *field)
 #if defined(J9VM_OPT_VALHALLA_FLATTENABLE_VALUE_TYPES)
 						J9UTF8 *fieldSig = J9ROMFIELDSHAPE_SIGNATURE(field);
 						U_8 *fieldSigBytes = J9UTF8_DATA(fieldSig);
-						if ('Q' == *fieldSigBytes) {
+						if ('Q' == *fieldSigBytes || J9_ARE_ALL_BITS_SET(modifiers, J9FieldFlagIsNullRestricted)) {
```
Comment (hangshao0) [inline runtime/vm/resolvefield.cpp:1108]: Please put the 2 conditions in separate lines:
```
if (('Q' == *fieldSigBytes)
    || J9_ARE_ALL_BITS_SET(modifiers, J9FieldFlagIsNullRestricted)
) {
```

File: test/functional/Valhalla/src_lw5/org/openj9/test/lworld/ValueTypeTests.java (hunk 1 above)
Comment (theresa-m) [reply test/functional/Valhalla/src_lw5/org/openj9/test/lworld/ValueTypeTests.java:45]: Sure that works.

File: runtime/vm/resolvefield.cpp [hunk 3]
```diff
@@ -1105,7 +1105,9 @@ fieldOffsetsFindNext(J9ROMFieldOffsetWalkState *state, J9ROMFieldShape *field)
 #if defined(J9VM_OPT_VALHALLA_FLATTENABLE_VALUE_TYPES)
 						J9UTF8 *fieldSig = J9ROMFIELDSHAPE_SIGNATURE(field);
 						U_8 *fieldSigBytes = J9UTF8_DATA(fieldSig);
-						if ('Q' == *fieldSigBytes) {
+						if ('Q' == *fieldSigBytes
```
Comment (hangshao0) [inline runtime/vm/resolvefield.cpp:1108]: missing `(` `)` around `'Q' == *fieldSigBytes`

Comment (theresa-m) [reply runtime/vm/resolvefield.cpp]: Updated.

**Review by hangshao0 — APPROVED** [review]

**Comment by hangshao0** [comment]
Jenkins test sanity,extended winval jdknext

**Comment by hangshao0** [comment]
Jenkins test sanity,extended zlinuxval jdknext

**Comment by hangshao0** [comment]
The failure in the PR [builds](https://openj9-jenkins.osuosl.org/job/Build_JDKnext_s390x_linux_valhalla_Personal/18/) looks related to API changes in the class library the JDKnext. https://github.com/eclipse-openj9/openj9/pull/18296 added the support in JDK22, but Valhalla is still failing. @JasonFengJ9

**Comment by JasonFengJ9** [comment]
JDK head stream merging conflict PRs were just merged today which the Valhalla repo depends on.
* https://github.com/ibmruntimes/openj9-openjdk-jdk/pull/676
* https://github.com/ibmruntimes/openj9-openjdk-jdk/pull/678

Will resolve Valhalla merging conflicts to consume the latest upstream and JDK head stream.

**Comment by hangshao0** [comment]
Jenkins test sanity,extended winval jdknext

**Comment by hangshao0** [comment]
Jenkins test sanity,extended zlinuxval jdknext

**Pull request merged**